wordpress-mobile/release-toolkit
# Feature/check translations status

This PR adds a new action which checks the state of the translations on GlotPress. 
It can be used to verify that the expected translations are available on GlotPress before finalising a release. 

Since there's no API on GlotPress (AFAIK) to get this information, but it's pretty easy to parse the HTTP code, the action extract the important data from the GlotPress project page. 

By default the action expects all the languages to be translated at 100% and raises a blocking error if it finds something below the threshold, but it can has options to modify the threshold and to ask user confirmation instead of failing directly. It has also an option to print the error report without blocking the flow at all. 

### To Test
- Verify CI is green. 
- Test on a live GlotPress repository (e.g. https://translate.wordpress.org/projects/apps/android/dev/) and verify the following options: 
a. _min_acceptable_translation_percentage_: pick a value that should trigger the error _and_ one which shouldn't (you can check the actual state opening the GlotPress project page on a web browser. 
b. _abort_on_violations = false_: set proper values of _min_acceptable_translation_percentage_ and verify that the output report is coherent and that the flow gets aborted when the user answers with **No** to the confirmation request. 
c. _abort_on_violations = false_ && _skip_confirm=true_: set proper values of _min_acceptable_translation_percentage_ and verify that the output report is coherent and that the flow is not interrupted.
